rhtconsulting/rhc-ose
# First cut at DNS ansible roles

#### What does this PR do?

Ansible roles for setting up a DNS server (bind/named) and allow for static DNS records to be added through the dns role (+ enables "nsupdate" for dynamic updates)
#### How should this be manually tested?

Use the files from the test directories to run the roles - i.e.:
- Update the dns-server role's test file with your desirable data for the dns-server
- run the dns-server role to configure the server

```
>> cd rhc-ose-ansible
>> cp roles/dns-server/test/role.yml .
>> [edit role.yml if desirable]
>> ansible-playbook role.yml
```
- run the dns role to add / remove / update static records

```
>> [cd rhc-ose-ansible]
>> cp roles/dns/test/records.yml .
>> [edit records.yml if desirable / to match the server config]
>> ansible-playbook records.yml
```
- use "dig" to test that the records are correctly populated - i.e.: if on a server with one ip, you can use the loopback address as the secondary view. (make sure the ACLs are correctly populated in the dns-server role to match your non-loopback IP for one of the views)

```
>> dig @<your_ip> <dns_name>
>> dig @127.0.0.1 <dns_name>
```
- Use "nsupdate" to add new "dynamic" records. i.e.: if on the actual server

```
>> nsupdate -k /var/named/<view>-<dns-domain>.key
server 127.0.0.1 # (or use the local ip)
update add <FQDN> 100 A <ip>
send
```
#### Is there a relevant Issue open for this?

N/A
#### Who would you like to review this?

/cc @etsauer @JayKayy @vvaldez @sabre1041
